atomix/cluster/src/main/java/io/atomix/raft/storage/log/DelayedFlusher.java

```diff
@@ -39,37 +38,33 @@ public DelayedFlusher(final Scheduler scheduler, final Duration delayTime) {
 
   @Override
   public void flush(final Journal journal) {
-    scheduleFlush(journal);
+    synchronized (scheduledMonitor) {
```

Comment:
I went this way, as AFAIK the performance difference between synchronized and reentrant locks is negligible, and in all cases I want to lock/unlock directly, so the structured approach is safer.